GSA-TTS/FAC
# 3695 address historical finding references with invalid formats

## Description
This PR addresses the issue of invalid reference number formats in historical data. For context, some historical audit reports failed the first round of data migration due to an invalid reference number format. Upon investigation, it turned out that the failures were due to the validation schema of the reference numbers, which only allowed the format YYYY-###. Here, YYYY must be a year greater than 2010, and ### represents any three digits. The error details revealed that the failing reports had reference numbers with years prior to 2010, such as 2006-010. The team-approved solution is to allow any year in the reference number.
This PR:
1. Update the reference number schema to allow any year greater than 1899.
2. Update the in-workbook validation rule to accommodate the new schema.
3. Update the workbook version and the approved workbook version list to include the new version.
4. Update existing test data fixtures to reflect the schema update.
5. Improve the intake validation process by adding a check for valid reference number to the corrective action plan and the finding text IRs validation (this is out of scope for historical data migration)

## How to test
Try migrating the report mentioned in #3695 to see if the invalid reference number error persist.

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
